src/generate.cc

```diff
@@ -980,11 +939,11 @@ static void place_random_door(int y, int x)
 		cave_set_feat(y, x, FEAT_BROKEN);
 	}
 
-	/* Secret doors (200/1000) */
+	/* Secret doors (200/1000) - now locked instead */
 	else if (tmp < 600)
 	{
-		/* Create secret door */
-		place_secret_door(y, x);
```

Comment:
Are secret doors on fixed levels, e.g. Death Watch, also handled correctly? (I.e. changed to locked doors.)